### PR TITLE
Update rules_go to v0.39.0 and gazelle to v.0.30.0

### PR DESCRIPTION
**Pull request by iain-macdonald**

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -20,35 +20,35 @@ http_archive(
...
     urls = [
-        "https://mirror.bazel.build/github.com/bazelbuild/bazel-gazelle/releases/download/v0.29.0/bazel-gazelle-v0.29.0.tar.gz",
-        "https://github.com/bazelbuild/bazel-gazelle/releases/download/v0.29.0/bazel-gazelle-v0.29.0.tar.gz",
+        "https://mirror.bazel.build/github.com/bazelbuild/bazel-gazelle/releases/download/v0.30.0/bazel-gazelle-v0.30.0.tar.gz",
+        "https://github.com/bazelbuild/bazel-gazelle/releases/download/v0.30.0/bazel-gazelle-v0.30.0.tar.gz",
     ],
 )
 
 load(":deps.bzl", "install_buildbuddy_dependencies")
+load("@io_bazel_rules_go//go:deps.bzl", "go_download_sdk", "go_register_toolchains", "go_rules_dependencies")
```
Comment (vadimberezniker) [inline WORKSPACE:42]: Do you remember the reason why you needed to move this?
The previous order was intentional (see the comment below).

**Review by vadimberezniker — CHANGES_REQUESTED** [review]

File: WORKSPACE (hunk 1 above)
Comment (iain-macdonald) [reply WORKSPACE]: Don't remember and managed to get it working without the ordering change, so reverted.

File: WORKSPACE [hunk 2]
```diff
@@ -80,9 +80,10 @@ go_download_sdk(
 
 go_register_toolchains(
     nogo = "@//:vet",
+    version = "1.18",
 )
 
-gazelle_dependencies()
+gazelle_dependencies(go_sdk = "go_sdk")
```
[the remark this answers is not in the document]
Comment (iain-macdonald) [reply WORKSPACE]: Per discussion, updated to version with Son' fix.

File: WORKSPACE [hunk 3]
```diff
@@ -20,21 +20,21 @@ http_archive(
 
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "dd926a88a564a9246713a9c00b35315f54cbd46b31a26d5d8fb264c07045f05d",
+    sha256 = "b18e85b0d6686f5072752cb5adc98f9a44efeb9f02181a59e040e092017e58c2",
+    strip_prefix = "rules_go-d756ad91feb9ca43800e781ab29c117623abed90",
     urls = [
-        "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.38.1/rules_go-v0.38.1.zip",
-        "https://github.com/bazelbuild/rules_go/releases/download/v0.38.1/rules_go-v0.38.1.zip",
+        "https://github.com/bazelbuild/rules_go/archive/d756ad91feb9ca43800e781ab29c117623abed90.tar.gz",
```
Comment (vadimberezniker) [inline WORKSPACE:26]: can you add a comment explaining why we're on this version and a note that we should upgrade to a point version once there's one with the fix

**Review by vadimberezniker — APPROVED** [review]

**Review by sluongng — APPROVED** [review]
Looks a lot cleaner than my https://github.com/buildbuddy-io/buildbuddy/pull/3699

Thanks!

**Pull request merged**